bridge-adapters/kinetic-bridgehub-adapter-generic-rest/src/main/java/com/kineticdata/bridgehub/adapter/generic/rest/GenericRestApiHelper.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.kineticdata.bridgehub.adapter.generic.rest;
+
+import com.kineticdata.bridgehub.adapter.BridgeError;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import net.minidev.json.JSONValue;
+import net.minidev.json.parser.ParseException;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GenericRestApiHelper {
+    private static final Logger LOGGER = 
+        LoggerFactory.getLogger(GenericRestApiHelper.class);
+    
+    private final String origin;
+    private final String username;
+    private final String password;
+    
+    public GenericRestApiHelper(String origin, String username, String password) {
+        
+        this.origin = origin;
+        this.username = username;
+        this.password = password;
+    } 
+    
+    public Object executeRequest (String partialUrl) throws BridgeError{
+        
+        Object output;      
+        // System time used to measure the request/response time
+        long start = System.currentTimeMillis();
+        
+        try (
+            CloseableHttpClient client = HttpClients.createDefault()
+        ) {
+            HttpResponse response;
+            HttpGet get = new HttpGet(origin + partialUrl);
+
+            get.setHeader("Authorization", getAuthHeader());
+            get.setHeader("Content-Type", "application/json");
+            get.setHeader("Accept", "application/json");
+            
+            response = client.execute(get);
+            LOGGER.debug("Recieved response from \"{}\" in {}ms.",
+                partialUrl,
+                System.currentTimeMillis()-start);
+
+            int responseCode = response.getStatusLine().getStatusCode();
+            LOGGER.trace("Request response code: " + responseCode);
+            
+            HttpEntity entity = response.getEntity();
+            // Confirm that response is a JSON object
+            output = parseResponse(EntityUtils.toString(entity));
+            
+            // Handle all other faild repsonses
+            if (responseCode >= 400) {
+                handleFailedReqeust(responseCode);
+            } 
+        }
+        catch (IOException e) {
+            throw new BridgeError("Unable to make a connection to the REST"
+                + " Service", e);
+        }
+        
+        return output;
+    }
+    
+    private void handleFailedReqeust (int responseCode) throws BridgeError {
```

Comment:
No biggie, but method is misspelled.

It would probably be beneficial to include the response message in the BridgeError for the 4xx errors since these are errors that the user can recover from by sending different data. The error usually contains information about what was wrong with the request.